### DISCUSSION TITLE
Has anyone here ever setup a strapi app on a cpanel ( we trying with nodejs. )

### DISCUSSION THREAD

**Discussion by Nemzytch** [strapi/strapi#7996] [Legacy Topics]
We've been trying to setup and run a strapi app on a  cpanel node js, we manage to finish the installation but can't get it to run properly, we ran into severall issues and are looking for someone who already achieved it totally. We are willing to pay for it as we need it professionnally. 

Thanks a lot in advance.

**Comment by derrickmehaffy** [comment]
While I've never used CPanel myself, can you describe what issues you are seeing (error messages from the server).

Note that if you are using shared hosting (multiple customers on a single server, in which you do not have admin level permissions over the server itself) you will likely be unable to deploy Strapi there due to the need to bind to ports.

**Reply by Nemzytch** [reply]
I'll look for a workaround later then, thanks for your time Derrick. 👍

**Reply by ziggybaba1** [reply]
Yes have set it up on a nodeJs server. You can mail me here seyiadejugbagbe1@gmail.com will love to help.

**Reply by abdelsalamshahlol** [reply]
@ziggybaba1 Can you share the method here?

**Reply by Nemzytch** [reply]
actually in the end we had put it directly on a VPS wich was wayyy easier. SO we can finally run in in DEV environnemnt, so we can still create new collection type, even in production.... I Still dont understand why strapi wouldnt allow it from start when you use something like heroku or whatever. They said it's for " security issues " but aren't we big enough to judge it ourself ? zzz

**Reply by abdelsalamshahlol** [reply]
Actually, I was able to run Strapi on shared hosting but I wouldn't recommend that due to performance issues. I'm getting a VPS for my client.

**Reply by Nemzytch** [reply]
Same for us ! And as I said, the good side of a VPS is that you will be able to run it easily in Developp mod rather than production mod, so your client, can still add new collection type/ elements etc. ( Wich is normally forbidden by strapi but it's like the main feature.. )

**Reply by derrickmehaffy** [reply]
@Nemzytch we don't allow it in production/staging because model settings are saved to files (hence why it won't work on most PaaS such as Heroku, GAP, DO Apps, ect as they generally have read-only file systems or they wipe the file system between deployments).

Our main reason behind it is to hopefully push the users into a proper code development flow with source control and cut down on the "test in production" which can easy cause issues and downtime for production applications.

If you want to read more: https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#why-can-t-i-create-or-update-content-types-in-production-staging
